plugins/catalog-backend/config.d.ts

```diff
@@ -216,5 +216,10 @@ export interface Config {
      * This flag is temporary and will be enabled by default in future releases.
      */
     useUrlReadersSearch?: boolean;
+
+    /**
+     * Evicts entities from the catalog when their related entity provider no longer exists.
+     */
+    evictOrphanedEntityProviders?: boolean;
```

Comment:
Potential alternative here: should `orphanStrategy: 'delete'` be the switch instead? I think it fits pretty well but is of course a risk at it affects existing users more directly. Probably want to avoid it because of that, but figured it's worth a discussion.

Reply:
I can imagine wanting finer grain control over orphans. 

It's reasonable to expect you might want to keep orphaned entities as they have accidentally been deleted in the source as that is out of your direct control. However, you have control over the providers configured, so would want to remove entities for providers that are no longer configured. 

I do prefer this configuration name though, so I might change it to `orphanedProviderStrategy: 'delete'`

Reply:
I've pushed config proposal [077493b](https://github.com/backstage/backstage/pull/28971/commits/077493b896e45a48c42b66951891b761e178a92d)